static/menu.js

```diff
@@ -1,9 +1,66 @@
+class SubMenuPositioner {
+    constructor(currentMenuSupplier) {
+        this.currentMenuSupplier = currentMenuSupplier;
+    }
+
+    getSubMenu(menu) {
+        return menu.getElementsByClassName('pure-menu-children')[0];
+    }
+
+    computeSubMenuOffset(menu) {
+        const { x: menuLeft } = menu.getBoundingClientRect();
+        const subMenu = this.getSubMenu(menu);
+
+        // Temporarily include the sub-menu in page layout in order to get the width
+        if ("block" !== window.getComputedStyle(subMenu).display) {
+            subMenu.style.display = "block";
+            subMenu.style.visibility = "hidden";
```

Comment:
Maybe the order here should be flipped to guarantee the sub-menu isn't visible before setting its display to block? Unsure if there's an actual possibility of a flicker here or not.